homeassistant/components/landisgyr_heat_meter/__init__.py

```diff
@@ -53,3 +53,38 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         hass.data[DOMAIN].pop(entry.entry_id)
 
     return unload_ok
+
+
+async def async_migrate_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Migrate old entry."""
+    _LOGGER.debug("Migrating from version %s", config_entry.version)
+
+    # Removing domain name and config entry id from entity unique id's, replacing it with device number
+    if config_entry.version == 1:
+
+        config_entry.version = 2
+
+        device_number = config_entry.data["device_number"]
```

Comment:
Is this the serial number?

Reply:
This is a unique number for the device, so sort of a serial number yes.